source/2_platform/rex_windows/src/entrypoint.cpp

```diff
@@ -25,20 +28,38 @@ INT APIENTRY WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPTSTR lpCmdL
 
   rex::ApplicationCreationParams app_params = rex::app_entry(rsl::move(creation_params), rsl::move(cmd_args));
 
-  // this doesn't initialize anything but simply prepares the application for initialization
-  rex::win32::GuiApplication application(rsl::move(app_params));
+  s32 result = 0;
+  if(app_params.create_window)
+  {
+    // this doesn't initialize anything but simply prepares the application for initialization
+    rex::win32::GuiApplication application(rsl::move(app_params));
 
-  // this initializes, runs the loop and performs the shutdown
-  s32 result = application.run();
+    // this initializes, runs the loop and performs the shutdown
+    result = application.run();
+  }
+  else
+  {
+    // this doesn't initialize anything but simply prepares the application for initialization
+    rex::win32::ConsoleApplication application(rsl::move(app_params));
+
+    // this initializes, runs the loop and performs the shutdown
+    result = application.run();
+  }
 
   // by this point the application has finished and shutdown
   REX_LOG(LogWindows, "Application completed with result: {0}", result);
 
   return result;
 }
 
+// main is always the entry point.
+// on a graphical application however, we call into WinMain
```

Comment:
We also call into win main on a none graphical application following this code.

Reply:
a comment has been added that console applications also use winmain